util/tls/tls_test.go

```diff
@@ -390,24 +392,66 @@ func TestCreateServerTLSConfig(t *testing.T) {
 	})
 }
 
+// getCert does the same thing as tls.AppendCertsFromPEM, but throws an error if something goes wrong.
+func getCert(pemCerts []byte) (*x509.Certificate, error) {
```

Comment:
golangci-lint was failing because `p.Subjects` was deprecated. This change lets us do the same test by using `p.Equal` against the expected cert pool instead of directly inspecting `p.Subjects`.